src/backend/access/heap/heapam.c

```diff
@@ -1395,8 +1395,12 @@ heap_endscan(TableScanDesc sscan)
 	pfree(scan);
 }
 
+/*
+ * NB: We no longer use heap_getnext() directly, please use
+ * table_scan_getnext() instead.
+ */
 HeapTuple
-heap_getnext(TableScanDesc sscan, ScanDirection direction)
```

Comment:
keep `heap_getnext` and just print warning `no longer use heap_getnext() directly`, that's better ?

Reply:
make sense.